executable-monitor/executable-monitor.py

```diff
@@ -41,6 +41,10 @@
                         type=int,
                         required=False,
                         help='Exit status that indicates that the executable completed successfully. Required if --success-line is not used.')
+    parser.add_argument('--retry-attempts',
```

Comment:
Not a fan of having retry attempts built in here. We should require that checks running in CI run reliably every time.

Reply:
It's actually entirely there for if doing an external network test where a failure due to not being able to connect is something that could happen. Where thought that having the ability to retry for something like that might be valuable in the future.